beacon-chain/slasher/receive.go

```diff
@@ -27,9 +27,34 @@ func (s *Service) receiveAttestations(ctx context.Context) {
 				Source:           att.Data.Source.Epoch,
 				Target:           att.Data.Target.Epoch,
 			}
-			s.queueLock.Lock()
+			s.attestationQueueLock.Lock()
 			s.attestationQueue = append(s.attestationQueue, compactAtt)
-			s.queueLock.Unlock()
+			s.attestationQueueLock.Unlock()
+		case err := <-sub.Err():
+			log.WithError(err).Debug("Subscriber closed with error")
+			return
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+// Receive beacon blocks from some source event feed,
+func (s *Service) receiveBlocks(ctx context.Context) {
+	sub := s.serviceCfg.BeaconBlocksFeed.Subscribe(s.beaconBlocksChan)
+	defer close(s.beaconBlocksChan)
+	defer sub.Unsubscribe()
+	for {
+		select {
+		case blockHeader := <-s.beaconBlocksChan:
+			// TODO(#8331): Validate block and defer blocks from the future for later processing.
```

Comment:
We dont need to validate the block for it to become slashable. A proposer can sign duplicated bad blocks and still be slashed

Reply:
We just need to ensure that this passes

https://github.com/ethereum/eth2.0-specs/blob/dev/specs/phase0/beacon-chain.md#proposer-slashings